server/mqtt_test.go

```diff
@@ -2100,7 +2086,7 @@ func TestMQTTParsePub(t *testing.T) {
 	}
 }
 
-func TestMQTTParsePubAck(t *testing.T) {
```

Comment:
You may want to modify the test above (`TestMQTTParsePub`) to update the `flags` value to `0x6` and the expected error message for the `qos not supported` test case, from `not supported` to your new error message `is invalid in MQTT`.